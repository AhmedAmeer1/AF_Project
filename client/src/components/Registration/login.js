import React, { PureComponent } from 'react'
import axios from 'axios'

class Login extends PureComponent {
    constructor(props) {
        super(props)

         
            this.onChangeEmail=this.onChangeEmail.bind(this);
            this.onChangePassword=this.onChangePassword.bind(this);

            this.onSubmit=this.onSubmit.bind(this);
        this.state = {
       
            email:'',
            password:'',
            loginUser:''
            
        }
    }




    onChangeEmail(e){
        this.setState({
            email: e.target.value
        })
    }


    onChangePassword(e){
        this.setState({
            password:e.target.value
        })
    }

    onSubmit(e){
        e.preventDefault();

        try {
                const users={
                   
                    email:this.state.email,
                    password:this.state.password
                }
                console.log(users);

                axios.post('http://localhost:5000/user/login',users).then(res =>{this.setState({loginUser:res.data})})

                
                axios.post('http://localhost:5000/user/login',users).then(res =>
                {
                    res.data.role==='user'? this.props.history.push('/userUI')
                    :
                    res.data.role==='admin'?  this.props.history.push('/editorUI')
                    :
                    this.props.history.push('/editorUI')
                
                
                })
                    
               

               
                


// //////////
// this.state.users.map(
//     user =>
//     console.log(user)
  
   
//     )
// /////////


                console.log("loginnnn : ")
                console.log(this.state.loginUser.role)

                if(this.state.loginUser.role === 'user'){
                    //this.props.history.push('/payment')
                }else{
                    //this.props.history.push('/ImageView')
                }

                


        } catch (error) {
            alert(error.response.data.msg)
        }
    }

    render() {
        return (
            <div>
                <h1>LOGIN FORM </h1>

            <form  onSubmit={this.onSubmit}>

            
              <p>Email  <input type='text'  name="email"  value={this.state.email}  onChange={this.onChangeEmail}/></p>
              <p>Password  <input type='text'  name="password"  value={this.state.password}  onChange={this.onChangePassword}/></p>

                <input type='submit' value="submit" />


            </form>

            </div>
        )
    }
}

export default Login