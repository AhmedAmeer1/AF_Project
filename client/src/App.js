
import './App.css';
import {BrowserRouter as Router ,Route, Switch} from "react-router-dom";

import Registration from './components/Registration/registration';


import Home from './components/home/home';
import UpdateRegistration from './components/admin/updateuser';
import Clouduploading from './components/imageUploading/clouduploading';
import ImageView from './components/imageUploading/imageView';
import Payment from './components/payment/payment';
import AboutUs from './components/aboutUs/aboutUs';
import Login from './components/Registration/login';







import ViewWorkshop from "./components/Users/viewWorkshop";


// user imports
import UserUI from './components/Users/userUI';
import CreateWorkshop from './components/Users/createWorkshop/createWorkshop';
import CreateResearchPaper from './components/Users/createResearchPaper/createResearchPaper';




// editor imports
import EditorUI from './components/Editors/editorUI';
import EditorViewConference from './components/Editors/editorViewConference';
import AddConference from './components/Editors/createConference/createConferece';
import EditConference from './components/Editors/editorUI';

// admin imports
import AdminUI from './components/admin/adminUI';
import AdminViewConference from './components/admin/viewConference';
import AdminViewUsers from './components/admin/adminViewUsers';
import AdminViewResearchPaper from './components/admin/adminViewResearchPaper';
import AdminViewWorkshop from './components/admin/AdminviewWorkshop';




// reviewer imports
import ReviwerUI from './components/reviewer/reviewerUI';
import ReviewerViewWorkshop from './components/reviewer/reviewerViewWorkshop';









function App() {
  return (
    <div className="App">
      
     
        
      <Router>
    

<div className="container">
    <Switch>
        <Route path="/"  exact component={Home}></Route>
        <Route path="/registration" component={Registration}></Route>
        <Route path="/updateUser/:_id" component={UpdateRegistration}></Route>
        <Route path="/img" component={Clouduploading}></Route>
        <Route path="/ImageView" component={ImageView}></Route>
        <Route path="/payment" component={Payment}></Route>
        <Route path="/login" component={Login}></Route>
        <Route path="/aboutUs" component={AboutUs}></Route>


        {/*user route*/}
        <Route path="/userUI" component={UserUI}></Route>
        <Route path="/addWorkshop/:email" component={CreateWorkshop}></Route>
        <Route path="/addWorkshop" component={CreateWorkshop}></Route>
        <Route path="/viewWorkshop" component={ViewWorkshop}></Route>
        <Route path="/createResearchPaper" component={CreateResearchPaper}></Route>





       {/*admin route*/}
        <Route path="/adminUI" component={AdminUI}></Route>
        <Route path="/adminViewConference" component={AdminViewConference}></Route>
        <Route path="/admin-view" component={AdminViewUsers}></Route>

        <Route path="/adminViewResearchPaper" component={AdminViewResearchPaper}></Route>
        <Route path="/adminViewWorkshop" component={AdminViewWorkshop}></Route>


        {/*reviewer route*/}
        <Route path="/reviwerUI" component={ReviwerUI}></Route>
        <Route path="/reviewerViewWorkshop" component={ReviewerViewWorkshop}></Route>



        {/*editor route*/}
        <Route path="/editorUI" component={EditorUI}></Route>
        <Route path="/addConference" component={AddConference}></Route>
        <Route path="/editConference" component={EditConference}></Route>
        <Route path="/editorViewConference" component={EditorViewConference}></Route>




        
        
        


                 



    </Switch>
</div>


</Router>
      
      
    </div>
  );
}

export default App;
