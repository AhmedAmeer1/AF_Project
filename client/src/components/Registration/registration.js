import React, { PureComponent } from 'react'
import Myheader from "../header/myheader";
import axios from 'axios'
import './login.css'




class Registration extends PureComponent {

    constructor(props) {
        super(props)

            this.onChangeName=this.onChangeName.bind(this);
        this.onChangeRole=this.onChangeRole.bind(this);
            this.onChangeEmail=this.onChangeEmail.bind(this);
            this.onChangePassword=this.onChangePassword.bind(this);

            this.onSubmit=this.onSubmit.bind(this);
        this.state = {
            name:'',
            role:'',
            email:'',
            password:''
            
        }
    }


    onChangeName(e){
        this.setState({
            name: e.target.value
        })
    }


    onChangeRole(e){
        this.setState({
            role: e.target.value
        })
    }




    onChangeEmail(e){
        this.setState({
            email: e.target.value
        })
    }


    onChangePassword(e){
        this.setState({
            password:e.target.value
        })
    }

    onSubmit(e){
        e.preventDefault();

        try {
                const users={
                    userRole:this.state.role,
                    name:this.state.name,
                    email:this.state.email,
                    password:this.state.password
                }
                console.log(users.role);

                axios.post('http://localhost:5000/user/register',users).then(res => alert(res.data.msg),

            users.role === 'workshop presenter' ? this.props.history.push(`/addWorkshop/${users.email}`)
            :
            this.props.history.push('/viewWorkshop')

            )



        } catch (error) {
            alert(error.response.data.msg)
        }
    }

    render() {
        return (

        <div>

    <Myheader/>
    <br/>
    <br/>
            <div className="login-page"style={{  height: 600, borderColor:"red"}}>
               
               
                <h1>Registration Form</h1>

            <form  onSubmit={this.onSubmit}   > 

            <div className="form-group" style={{borderColor:"red"}}>
                    
                    <input type="text" className="form-control" style={{borderColor:"red"}} placeholder="Enter Name" name="name" value={this.state.name} onChange={this.onChangeName} />
            </div>    

           
                     <div className="form-group">
                     <select    name="role" value={this.state.role} onChange={this.onChangeRole}>
                        <option value="attendee" >attendee</option>
                        <option value="researcher">researcher</option>
                        <option value="workshop presenter">workshop presenter</option>
                    </select>
                     </div>

              
     
            <div className="form-group">
                   
                    <input type="email" className="form-control" style={{borderColor:"#FF0000"}} placeholder="Enter Email" name="email"  value={this.state.email}  onChange={this.onChangeEmail} />
            </div>



               

               <div className="form-group">
                   
                    <input type="text" className="form-control" style={{borderColor:"#FF0000"}} placeholder="Enter Password" name="password"  value={this.state.password}  onChange={this.onChangePassword} />
            </div>

              <div className="row">
              <button type="submit" style={{backgroundColor:"#DC3545", border:"none", marginLeft:"20px"}}>submit</button>
              </div>
            </form>

            </div>

        </div>
        )
    }
}

export default Registration