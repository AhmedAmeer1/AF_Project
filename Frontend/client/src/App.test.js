import { render, screen } from '@testing-library/react';
import App from './App';
import React from 'react'
import{shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import fetchMock from 'fetch-mock'
import UserUI from "../../../client/src/components/Users/userUI";
import Registration from "./components/registration";

//jest snapshot
describe('APP component',()=>{
  it('matches the snapshot of app component',()=>{
    const tree =renderer.create(<App/>).toJSON

    //expect
    expect(tree).toMatchSnapshot()
  })
})
const url="dsdsd"
const dummyUser={
  name:"sdsdd"
}


// API FETCH TEST
/*
descripe('GET VALUE FROM USER ID'),()=>{
  it('shows the name of the user after fetched ',async ()=>{
     fetchMock.getOnce(url,{
       status:200,
       body:dummyUser
     })
    const wrapper = shallow(<UserUI id={1}/>)
    await nextTick()
    expect(wrapper.find('p').first).toEqual(dummyUser.name)

  })
})
*/

//CHECK INITIAL VALUE OF THE FORM
describe(' CHECK INTIAL VALUES OF FORM',()=>{
  it('has initial value as empty VALUE',()=>{
    const wrapper =<Registration/>
    const name=wrapper.find('input').first
    expect(name.props().value).toEqual('')
    const email=wrapper.find('input').second
    expect(email.props().value).toEqual('')

  })
})


//CHECK ONCHANGE VALUE
const simulateChangeOnInput =(wrapper1,inputSelector,newValue)=>{
  const input =wrapper1.find(inputSelector)
  input.simulate('change',{
    target:{
      value:newValue
    }
  })
  return wrapper1.find(inputSelector)

}
describe('CHECK VALUE ON ONCHANGE IN REGISTRATION FORM',()=>{
  it("check the updated value onchange",()=>{
    const wrapper1 = shallow(<Registration/>)
    let nameInput=wrapper1.find('input').first
    const updatedNameInput =simulateChangeOnInput(wrapper1,'input#name','suman')
    expect(updatedNameInput.props().value).toEqual("suman")
  })

})