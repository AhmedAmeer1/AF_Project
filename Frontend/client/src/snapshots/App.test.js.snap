exports[`App component matches the snapshot 1`]=`  <div className="App">



                                                       <Router>
                                                       <Header />

                                                 <div className="container">
                                                     <Switch>
                                                         <Route path="/"  exact component={Home}></Route>
                                                         <Route path="/registration" component={Registration}></Route>
                                                         <Route path="/updateUser" component={UpdateRegistration}></Route>
                                                         <Route path="/admin-view" component={AdminViewUsers}></Route>
                                                         <Route path="/img" component={Clouduploading}></Route>
                                                         <Route path="/ImageView" component={ImageView}></Route>

                                                         ImageView



                                                     </Switch>
                                                 </div>


                                                 </Router>


                                                     </div>`;